app/controllers/concerns/lti_support.rb

```diff
@@ -83,6 +83,14 @@ def user_from_lti
       user
     end
 
+    # The User Tool requires the launching user to be an LTI admin in the current context.
+    def validate_user_is_admin_for_user_tool
+      if current_application_instance.lti_key == Application::USERTOOL
+        roles = params[:roles].split(",")
```

Comment:
I think the more proper method here is to use `#lti_roles` so we can catch when roles come in on the key `ext_roles`.
```suggestion
        roles = lti_roles.split(",")
```

Reply:
I considered using the `lti_roles` method, but using the roles given in `params[:ext_roles]` (if they exist) didn't seem to provide the behavior I wanted. I'm not confident on my understanding of the differences between `params[:roles]` and `params[:ext_roles]`, but what I observed was that when I attempted to launch the tool with a sub-account admin from a different sub-account than the one the tool was being launched from (someone who shouldn't have permission to use the tool) that sub-account admin had an admin role listed in `params[:ext_roles]`. However, they did not have an admin role listed in `params[:roles]`. When I launched the tool as a sub-account admin for the account the tool was being launched from, that user had an admin role listed in both `params[:ext_roles]` and `params[:roles]`. Also, the comment on line 166 of `lti_support.rb` seems to indicate that `params[:roles]` contains roles for the current context. 

Between those 2 observations, I concluded using `params[:roles]` was what we wanted in this case. If we used `lti_roles` and thus checked the roles from `params[:ext_roles]` it would allow anyone with an admin role to launch the tool even if they aren't an admin for the account from which the tool is being launched.

Reply:
Oh, fair enough. Let's roll with `params[:roles]` then.

Reply:
Ok, cool. Thanks.